ntoskrnl/inbv/inbv.c

```diff
@@ -1066,13 +1199,19 @@ DisplayBootBitmap(IN BOOLEAN TextMode)
              * Save the main image palette and replace it with black palette,
              * so that we can do fade-in effect later.
              */
-            BitmapInfoHeader = (PBITMAPINFOHEADER)Screen;
-            Palette = (LPRGBQUAD)((PUCHAR)Screen + BitmapInfoHeader->biSize);
+            BitmapInfoHeader = BootLogo;
+            Palette = (LPRGBQUAD)((PUCHAR)BootLogo + BitmapInfoHeader->biSize);
             RtlCopyMemory(MainPalette, Palette, sizeof(MainPalette));
             RtlZeroMemory(Palette, sizeof(MainPalette));
```

Comment:
I wonder why but this code should also BSOD, according to what I mentioned in a comment above.
I will investigate further.

Reply:
I'm not sure, but maybe it's due to the fact that I removed unnecessary casts?

Reply:
No, with or without the casts for BitmapInfoHeader, the code is going to write to the same memory address, so it should be caused by a different reason...

Reply:
The answer to my question is: "it BSODs", but it currently doesn't because it is run before Mm sets some extra memory protection. The new iteration of the commit explicitly changes the memory protection of the kernel resource section.